pay-api/src/pay_api/services/payment.py

```diff
@@ -366,21 +382,54 @@ def create_payment_report(auth_account_id: str, search_filter: Dict,
 
         results = Payment.search_all_purchase_history(auth_account_id, search_filter)
 
-        report_response = Payment.generate_payment_report(content_type, report_name, results,
-                                                          template_name='payment_transactions')
+        report_response = Payment.generate_payment_report(PaymentReportInput(content_type=content_type,
+                                                                             report_name=report_name,
+                                                                             template_name='payment_transactions',
+                                                                             results=results
+                                                                             ))
         current_app.logger.debug('>create_payment_report')
 
         return report_response
 
+    @staticmethod
+    def get_invoices_totals(invoices: List[InvoiceModel]):
+        """Tally up totals for a list of invoices."""
+        total_stat_fees = 0
+        total_service_fees = 0
+        total = 0
+        total_paid = 0
+
+        for invoice in invoices:
```

Comment:
I feel like we already do this somewhere else? maybe in NSF?  also weird we can't just use an invoice like an object EG. `invoice.paid` instead of` invoice.get('paid')`

Reply:
Usually when I see this, it is because previously paid potentially had null values and the model will just plain not have that attribute causing it to error.

This was existing logic I just moved into a function, it is possible NSF is doing this.